pkgs/misc/vscode-extensions/python/default.nix

```diff
@@ -14,16 +14,48 @@ assert ctagsUseFixed -> null != ctags;
 let
   pythonDefaultsTo = if pythonUseFixed then "${python}/bin/python" else "python";
   ctagsDefaultsTo = if ctagsUseFixed then "${ctags}/bin/ctags" else "ctags";
-in
 
-vscode-utils.buildVscodeMarketplaceExtension {
+  # The arch tag comes from 'PlatformName' defined here:
+  # https://github.com/Microsoft/vscode-python/blob/master/src/client/activation/types.ts
+  arch =
+    if stdenv.isLinux && stdenv.isx86_64 then "linux-x64"
+    else if stdenv.isDarwin then "osx-x64"
+    else throw "Only x86_64 Linux and Darwin are supported.";
+
+  languageServerSha256 = {
+    "linux-x64" = "0mqjl3l1zk1zd7n0rrb2vdsrx6czhl4irdm4j5jishg9zp03gkkd";
+    "osx-x64" = "1csq8q8fszv9xk9qiabg12zybxnzn8y2jsnvjrlg4b8kvm63sz40";
+  }."${arch}";
+
+  # version is languageServerVersion in the package.json
+  languageServer = extractNuGet rec {
+    name = "Python-Language-Server";
+    version = "0.2.82";
+
+    src = fetchurl {
```

Comment:
@worldofpeace : My understanding as to why `fetchNuGet` is not used is that there is no official standalone release yet as mentioned in [Add installation instructions to README · Issue #953 · microsoft/python-language-server](https://github.com/microsoft/python-language-server/issues/953#issuecomment-483067448).

 >  We do not provide official downloads for the installation yet (although download URLs are not secret but they can change). So it is recommended for clients to build their own. One of the reasons is that build we use is only tested with VS Code.

It is also **not possible** (yet) to build this be ourselves as *xbuild* is officially deprecated in favor of `msbuild` (see [Mono 5.0.0 Release Notes | Mono](https://www.mono-project.com/docs/about-mono/releases/5.0.0/#msbuild)) and does not meet the minimal requirement of this project (tried it myself).

Adding support for `msbuild` (even borrowed from debian) is an unfinished process:

 -  [mono50 doesn't include msbuild · Issue #29817 · NixOS/nixpkgs](https://github.com/NixOS/nixpkgs/issues/29817)
 -  [msbuild: init at 15.6 by leo60228 · Pull Request #43680 · NixOS/nixpkgs](https://github.com/NixOS/nixpkgs/pull/43680)
 -  [msbuild: init at 15.8 by leo60228 · Pull Request #50842 · NixOS/nixpkgs](https://github.com/NixOS/nixpkgs/pull/50842)
 
So to me, what's has been done here is fine for the time being.

Reply:
Yep completely agree here. Perhaps a todo should be added though for when that's resolved.